packages/extension-driver-ksqldb/src/lib/restfulClient.ts

```diff
@@ -0,0 +1,214 @@
+import * as http2 from 'http2';
+
+const BASIC_HEADERS = {
+  Accept: 'application/vnd.ksql.v1+json',
+};
+
+const RESTFUL_API = {
+  INFO: '/info',
+  KSQL: '/ksql',
+  QUERY: '/query',
+};
+
+interface KsqlInfoResponse {
+  KsqlServerInfo: {
+    version: string;
+    kafkaClusterId: string;
+    ksqlServiceId: string;
+    serverStatus: string;
+  };
+}
+
+export type Header = {
+  header: {
+    queryId: string;
+    schema: string;
+  };
+};
+
+export type Row = {
+  row: {
+    columns: any[];
+  };
+};
+
+export type FinalMessage = {
+  finalMessage: string;
+};
+
+/**
+ * The example query response like below:
+ * [
+ *  {
+ *    "header": {
+ *      "queryId": "transient_RIDERLOCATIONS_356492705638097482",
+ *      "schema": "`PROFILEID` STRING, `LATITUDE` DOUBLE, `LONGITUDE` DOUBLE"
+ *    }
+ *  },
+ *  { "row": {"columns": ["c2309eec",37.7877,-122.4205]} },
+ *  { ...more rows },
+ *
+ *  { "finalMessage": "Query Completed"}
+ * ]
+ */
+export type QueryResponse = Header | Row | FinalMessage;
+
+export interface RestfulClientOptions {
+  host: string;
+}
+
+export class RestfulClient {
+  private options: RestfulClientOptions;
+  public client?: http2.ClientHttp2Session;
+  public connected = false;
+  private startSession: () => null | Promise<void> = () => null;
+
+  constructor(options: RestfulClientOptions) {
+    this.options = options;
+    this.connect();
+  }
+
+  public connect() {
```

Comment:
Suggest to add comment to explain why to define `startSession` when calling `connect` but not do any actual connecting action

Reply:
I have added the comment in [03a2ffa](https://github.com/Canner/vulcan-sql/pull/255/commits/03a2ffaea6fa104d8d643e118ba1cb32bc13fa97).